### PR TITLE
feat-add-support-for-apiKey

### PR DESCRIPTION
**Pull request by ComeOnOliver**

### CODE REVIEW

File: src/libs/Aggregator/utils.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { createHash } from 'crypto';
```
Comment (alexayan) [inline src/libs/Aggregator/utils.ts:1]: browser environment does not have this module.

File: src/libs/Aggregator/utils.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    refIdCandidate = refIdCandidate % REF_ID_MOD;
+
+    // Avoid conflicts with reserved ref_id
+    let offset = 0;
+    let finalRefId = refIdCandidate;
+    // Try increasing offset each time and take modulo to stay within 10 digits
+    while (RESERVED_REF_IDS.has(finalRefId)) {
+        offset += 1;
+        finalRefId = (refIdCandidate + offset) % REF_ID_MOD;
+    }
```
Comment (alexayan) [inline src/libs/Aggregator/utils.ts:37]: Different reserved ids generate the same value.

File: src/libs/Aggregator/utils.ts (hunk 1 above)
Comment (ComeOnOliver) [reply src/libs/Aggregator/utils.ts]: fixed

File: src/libs/Aggregator/utils.ts (hunk 2 above)
Comment (ComeOnOliver) [reply src/libs/Aggregator/utils.ts:36]: This is to avoid [1873161113, 8190801341] that originally used in NAVI

**Review by alexayan — APPROVED** [review]

**Pull request merged**